### PR TITLE
Remove trailing whitespace

### PR DESCRIPTION
**Pull request by aconsroe-hpe**
This continues the work of #7848 by removing all trailing whitespace that would have been warned on and enables this check in smokeTest.

The dirs that are searched are the same as before: compiler, runtime, modules, and tools. See util/devel/lookForTrailingSpaces

Also pointed out by @vasslitvinov that there was a trailing whitespace in our internal error message that is now removed and the associated .bad files are updated.

* [x] paratest (w/ futures)

### CODE REVIEW

**Comment by ben-albrecht** [comment]
IIRC, there were several tests that expect trailing whitespace in the `.good` file -- you'll probably catch these in the paratest. Depending on how much effort those would require to update, we could consider ignoring this check for `.good`/`.bad` files.

**Comment by aconsroe-hpe** [comment]
Do you mean that there is a .good file with the `We would appreciate your reporting this bug -- ` that will now not match? I'm not seeing any of those. If you just mean in general that there are .good with trailing space, that is okay and will not be checked by the smokeTest

**Comment by ben-albrecht** [comment]
> If you just mean in general that there are .good with trailing space, that is okay and will not be checked by the smokeTest

This is what I meant. If those already aren't checked by the smokeTest, then we're good!

**Review by lydia-duncan — COMMENTED** [review]
Pointed out some regularly generated files that we shouldn't be checking with the script.  Also noted something that was generated but won't change in the future - the changes to the file itself are good, but it would good to track down if they can still get inserted by the program that created them (and stop it if so)

**Comment by lydia-duncan** [comment]
Thanks for taking this on!  It's disappointing there were so many changes needed but it'll be nice to having to deal with it going forward

File: compiler/next/lib/immediates/hash_multipliers.cpp [hunk 1]
```diff
@@ -22,36 +22,36 @@
 
 // primes generated with map_mult.c
 uint32_t open_hash_multipliers[256] = {
-0x02D4AF27, 0x1865DFC7, 0x47C62B43, 0x35B4889B, 0x210459A1, 0x3CC51CC7, 0x02ADD945, 0x0607C4D7,
```
[the remark this answers is not in the document]
Comment (aconsroe-hpe) [reply compiler/next/lib/immediates/hash_multipliers.cpp:25]: I don't anticipate this changing very often

File: compiler/include/flex-chapel.h [hunk 2]
```diff
@@ -62,7 +62,7 @@
 #if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 
 /* C99 says to define __STDC_LIMIT_MACROS before including stdint.h,
- * if you want the limit (max/min) macros for int types.
```
[the remark this answers is not in the document]
Comment (aconsroe-hpe) [reply compiler/include/flex-chapel.h:65]: Will remove the changes on these, but note that the tool `lookForTrailingSpaces` already skips these so no error will/would be generated

File: modules/packages/LAPACK.chpl [hunk 3]
```diff
@@ -19093,8295 +19093,8295 @@ pragma "no doc"
 extern proc LAPACK_ilaver(ref vers_major : c_int, ref vers_minor : c_int, ref vers_patch : c_int) : c_void_ptr;
 
 } // ClassicLAPACK
-/* 
+/*
```
[the remark this answers is not in the document]
Comment (aconsroe-hpe) [reply modules/packages/LAPACK.chpl]: I skimmed over c2chapel.py and didn't find an obvious culprit

File: compiler/next/lib/parsing/flex-chpl-lib.cpp [hunk 4]
```diff
@@ -268,7 +268,7 @@
 #if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 
 /* C99 says to define __STDC_LIMIT_MACROS before including stdint.h,
- * if you want the limit (max/min) macros for int types.
```
[the remark this answers is not in the document]
Comment (aconsroe-hpe) [reply compiler/next/lib/parsing/flex-chpl-lib.cpp:271]: this is not ignored by the existing script, will revert and modify the script to skip these

**Comment by aconsroe-hpe** [comment]
I backed out those lexer/parser generated files and updated the ignore list in `lookForTrailingSpaces` to account for compiler/next

**Review by lydia-duncan — APPROVED** [review]
Awesome, thanks!

**Pull request merged**